cyipopt/scipy_interface.py

```diff
@@ -1,239 +1,240 @@
-            options[newname] = options.pop(oldname)
-
-def convert_to_bytes(options):
-    if sys.version_info >= (3, 0):
-        for key in list(options.keys()):
-            try:
-                if bytes(key, 'utf-8') != key:
-                    options[bytes(key, 'utf-8')] = options[key]
-                    options.pop(key)
-            except TypeError:
-                pass
-
-def minimize_ipopt(fun, x0, args=(), kwargs=None, method=None, jac=None, hess=None, hessp=None,
-                   bounds=None, constraints=(), tol=None, callback=None, options=None):
-    """
-    Minimize a function using ipopt. The call signature is exactly like for
-    `scipy.optimize.mimize`. In options, all options are directly passed to
-    ipopt. Check [http://www.coin-or.org/Ipopt/documentation/node39.html] for
-    details.
-    The options `disp` and `maxiter` are automatically mapped to their
-    ipopt-equivalents `print_level` and `max_iter`.
-    """
-    if not SCIPY_INSTALLED:
-        raise ImportError('Install SciPy to use the `minimize_ipopt` function.')
-
-    _x0 = np.atleast_1d(x0)
-    problem = IpoptProblemWrapper(fun, args=args, kwargs=kwargs, jac=jac, hess=hess,
-                                  hessp=hessp, constraints=constraints)
-    lb, ub = get_bounds(bounds)
-
-    cl, cu = get_constraint_bounds(constraints, x0)
-
-    if options is None:
-        options = {}
-
-    nlp = cyipopt.Problem(n = len(_x0),
-                          m = len(cl),
-                          problem_obj=problem,
-                          lb=lb,
-                          ub=ub,
-                          cl=cl,
-                          cu=cu)
-
-    # python3 compatibility
-    convert_to_bytes(options)
-
-    # Rename some default scipy options
-    replace_option(options, b'disp', b'print_level')
-    replace_option(options, b'maxiter', b'max_iter')
-    if b'print_level' not in options:
-        options[b'print_level'] = 0
-    if b'tol' not in options:
-        options[b'tol'] = tol or 1e-8
-    if b'mu_strategy' not in options:
-        options[b'mu_strategy'] = b'adaptive'
-    if b'hessian_approximation' not in options:
-        if hess is None and hessp is None:
-            options[b'hessian_approximation'] = b'limited-memory'
-    for option, value in options.items():
-        try:
-            nlp.add_option(option, value)
-        except TypeError as e:
-            raise TypeError('Invalid option for IPOPT: {0}: {1} (Original message: "{2}")'.format(option, value, e))
-
-    x, info = nlp.solve(_x0)
-
-    if np.asarray(x0).shape == ():
-        x = x[0]
-
-    return OptimizeResult(x=x, success=info['status'] == 0, status=info['status'],
-                          message=info['status_msg'],
-                          fun=info['obj_val'],
-                          info=info,
-                          nfev=problem.nfev,
-                          njev=problem.njev,
-                          nit=problem.nit)
+#!/usr/bin/env python

```

Comment:
Do you have different line endings on this file? Git seems to be registering that all the lines are changed, instead of just the new ones.

Reply:
Strange, using VScode, I will check

Reply:
Arggh, set to CRLF, not LF, I am sure I have set this right earlier - then again I use Visual Studio for the only other project where I commit code to Git/GitHub, so I might have missed it. The diffs in VSCODE looked right, only highlighting lines with actual changes. What to do?

Reply:
See, but not cheking again I can see whole blocks with no changes also hightlighted as changes
![image](https://user-images.githubusercontent.com/58475535/104470191-bcd02380-55b9-11eb-8a1a-c5d67c63964f.png)

Reply:
You can set the git setting to convert line endings on commit: https://docs.github.com/en/free-pro-team@latest/github/using-git/configuring-git-to-handle-line-endings

Reply:
What about the fork/branch already in pull request?

Reply:
I think you'd make the setting and push a new commit. I should fix the files. But maybe you'd have to edit the file.

Reply:
I think we are getting there, awaiting the buil check

Reply:
Let me know  if more changes are required - sadly IPOPT wasn't ideal for my problem this time (black-box with numerical derivatives), but at least now I have it in the python toolbox, thanks a lot for making this interface.